src/Microsoft.AspNetCore.Blazor.Server/AutoRebuild/AutoRebuildExtensions.cs

```diff
@@ -77,6 +109,12 @@ public static void UseAutoRebuild(this IApplicationBuilder appBuilder, BlazorCon
             });
         }
 
+        private static string FindFirstFileInDirectoryRecursive(string rootDirectory, string pattern)
```

Comment:
What do you think of leveraging the new .Net Core 2.1 FileSystemEnumerable, that's more lightweight and offers plenty of options.

Something similar:

```
            var enumerationOptions = new EnumerationOptions
            {
                IgnoreInaccessible = true,
                RecurseSubdirectories = true
            };

            var fileToTouch = new FileSystemEnumerable<string>(hostAppContentRootPath, (ref FileSystemEntry entry) => entry.ToFullPath(), enumerationOptions)
            {
                ShouldIncludePredicate = (ref FileSystemEntry entry) => !entry.IsDirectory && entry.FileName.EndsWith(".cs", StringComparison.OrdinalIgnoreCase)
            };
 ```

Reply:
I'm sure that would work fine, but `Directory.EnumerateFiles` is even simpler and given the use case (invoked at most once per app build) there is no chance that any perf difference would be observable.